src/counters.c

```diff
@@ -778,6 +779,8 @@ static int StatsOutput(ThreadVars *tv)
             StatsRecord *r = &stats_table.tstats[offset];
             /* xfer previous value to pvalue and reset value */
             r->pvalue = r->value;
+            r->counter_pvalue = r->counter_value;
```

Comment:
do you really need counter_pvalue and counter_value? Feels like you can do that just with e/r.value and r.pvalue

Reply:
I have to motivation.

I used it because I want to keep r.value and r.pvaule with the original meaning, so using a STATS_TYPE_TIMED qualifier this values will keep the counter/sec and using a STATS_TYPE_AVERAGE this values will keep the counter average.

Furthermore, I think the value r->counter_p and r->counter_value are necessary because after the statistics window interval the v->value will be used in the next iteration. Over time the r->value will get distorted if I use something like:

r->value = (r->value - r->pvalue) / SEC

So I guess that is necessary a intermediary variable